core/src/unprocessed_packet_batches.rs

```diff
@@ -372,9 +374,19 @@ pub fn packet_message(packet: &Packet) -> Result<&[u8], DeserializedPacketError>
         .ok_or(DeserializedPacketError::SignatureOverflowed(sig_size))
 }
 
-/// Computes `(addition_fee + base_fee / requested_cu)` for `deserialized_packet`
-fn compute_fee_per_cu(_message: &SanitizedVersionedMessage, _bank: &Bank) -> u64 {
-    1
+/// Computes fee_per_cu as `(addition_fee / requested_cu)`
+fn compute_fee_per_cu(message: &SanitizedVersionedMessage) -> Option<f64> {
+    let mut compute_budget = ComputeBudget::default();
+    let additional_fee = compute_budget
+        .process_instructions(
+            message.program_instructions_iter(),
+            false, // not request heap size
+            true,  // use default units per instruction
+        )
+        .ok()?;
+    let additional_fee = f64::from_u64(additional_fee)?;
+    let request_units = f64::from_u64(compute_budget.max_units)?;
+    Some(additional_fee / request_units)
```

Comment:
Can we use integer arithmetic here? What's the benefit of using floating point over fixed point? If fee is 10,000 lamports and compute is 1,000,000 units, then we can compute the fee_per_cu by scaling up lamports (using 128) by some large factor (let's say 10^9) and then use integer division to get `micro_lamports_per_cu` as a `u64`.. so something like (10,000 * 10^9) / 1,000,000 = 10,000,000 micro lamports per cu.

Reply:
Looks like `compute-budget` will keep additional_fee as lamports per tx, which is totally fine. [Commit](fdfa89eb3ebdc96dfda1bf3463b518d2a9e0e7f7) implements the suggestion